yt_geotiff/io.py

```diff
@@ -113,11 +125,150 @@ def _read_fluid_selection(self, chunks, selector, fields, size):
                     nd = g.select(selector, data, rv[field], ind)
 
                 ind += nd
+    
+        return rv
+
+class io_handler_JPEG2000(IOHandlerGeoTiff):
+    _dataset_type = "JPEG2000"
+
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+
+        rv = {}
+        chunks = list(chunks)
+
+        if isinstance(selector, GridSelector):            
+
+            if not (len(chunks) == len(chunks[0].objs) == 1):
+                raise RuntimeError
+            g = chunks[0].objs[0]
+
+            if g.id in self._cached_fields:
+                gf = self._cached_fields[g.id]
+                rv.update(gf)
+
+            if len(rv) == len(fields):
+                return rv
+            
+            for field in fields:
+                if field in rv:
+                    self._hits += 1
+                    continue
+                self._misses += 1
+                ftype, fname = field 
+                  
+                filename=self.ds._field_filename[fname]['filename']
+
+                src = rasterio.open(filename, "r")
+                rasterio_window = g._get_rasterio_window(selector)
+
+                # round up rasterio window width and height
+                rasterio_window = rasterio_window.round_shape(op='ceil', pixel_precision=None)
+
+                # Read in the band/field
+                data = src.read(1, window=rasterio_window).astype(
+                    self._field_dtype)
+
+                rv[(ftype, fname)] = self._transform_data(data)
```

Comment:
I think we should save assignment to the return value data structure until the last step so as to signal that we are done modifying the data.
```suggestion
                data = self._transform_data(data)
```